packages/react-button/src/components/Button/useButton.ts

```diff
@@ -1,28 +1,34 @@
 import * as React from 'react';
-import { getStyleFromPropsAndOptions } from '@fluentui/react-theme-provider';
-import { useFocusRects } from '@uifabric/utilities';
-import { ComposePreparedOptions } from '@fluentui/react-compose';
-import { ButtonProps, ButtonState } from './Button.types';
+import { ButtonState } from './Button.types';
+// import { useMergedRefs } from '@uifabric/react-hooks';
 
 /**
- * The useButton hook processes the Button component props and returns state.
- * @param props - Button props to derive state from.
+ * The useButton hook processes the Button draft state.
+ * @param draftState - Button draft state to mutate.
  */
-export const useButton = (
-  props: ButtonProps,
-  ref: React.Ref<HTMLElement>,
-  options: ComposePreparedOptions,
-): ButtonState => {
-  const buttonRef = React.useRef<HTMLButtonElement | null>(null);
+export const useButton = (draftState: ButtonState) => {
+  // Update the button's tab-index, keyboard handling, and aria attributes.
+  if (draftState.as !== 'button') {
```

Comment:
Using `as` is not enough to determine `tagName`

Reply:
Better suggestion?

Reply:
https://github.com/reakit/reakit/blob/9bd6882fb50948c616e757d750b436decc36ddfc/packages/reakit/src/Button/Button.ts#L40
https://github.com/microsoft/fluentui/pull/13717#discussion_r454935803

Reply:
I think this is good enough; I would hate to add perf overhead calling `useEffect` to evaluate if we need to re-evaluate attributes and then re-rendering in this case.

In the worse case scenario, we find `{as: Foo}` and we treat it like a `span`. This technically is more work than needed but shouldn't be breaking.